### PR TITLE
IGAPP-719: Upgrade to react native 0.64.2 and react 17.0.2

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Comment by klinzo** [comment]
Please test this PR thoroughly including pdfs, images, feedback, webview,... 
Images and PDFs can be found here: https://integreat.app/testumgebung/de

**Comment by klinzo** [comment]
Known Issues:
- [ ] Text in location search is not visible

**Review by SteffiStuffel — COMMENTED** [review]
Could not install the app:

error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup.
Error: Command failed: ./gradlew app:installDebug -PreactNativeDevServerPort=8081

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* Where:
Build file '/home/steffi/Dokumente/integreat-app/native/android/app/build.gradle' line: 132

**Comment by klinzo** [comment]
> Could not install the app:
> 
> error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup.
> Error: Command failed: ./gradlew app:installDebug -PreactNativeDevServerPort=8081
> 
> FAILURE: Build completed with 2 failures.
> ## 1: Task failed with an exception.
> 
>     * Where:
>       Build file '/home/steffi/Dokumente/integreat-app/native/android/app/build.gradle' line: 132

I guess you have to run a `./gradlew clean` in `android`. Maybe deleting/cleaning other things is necessary as well. Should have mentioned that.

File: native/android/app/build.gradle [hunk 1]
```diff
@@ -129,6 +129,8 @@ def jscFlavor = 'org.webkit:android-jsc:+'
 def enableHermes = project.ext.react.get("enableHermes", false);
 
 android {
+    ndkVersion rootProject.ext.ndkVersion
```
Comment (Taggotty) [inline native/android/app/build.gradle:132]: Why did you add this. After removing the app compiles again,

**Review by Taggotty — APPROVED** [review]
Installation worked after removing the ndk version and deleting both build folders in the android package. Did not find any issues in the app.

File: native/android/app/build.gradle (hunk 1 above)
Comment (klinzo) [reply native/android/app/build.gradle:132]: https://react-native-community.github.io/upgrade-helper/?from=0.63.4&to=0.64.2
Upgrade helper says so. Seems like I forgot the `ndkVersion` in `android/build.gradle` though.

File: native/android/gradle.properties [hunk 2]
```diff
@@ -25,4 +25,4 @@ android.useAndroidX=true
 android.enableJetifier=true
 
 # Version of flipper SDK to use with React Native
-FLIPPER_VERSION=0.96.1
+FLIPPER_VERSION=0.75.1
```
[the remark this answers is not in the document]
Comment (klinzo) [reply native/android/gradle.properties:28]: Did not get it to work with higher versions and it is the recommended/supported version with react native 0.64.2.

File: native/ios/Integreat/Info.plist [hunk 3]
```diff
@@ -32,8 +32,6 @@
 	<true/>
 	<key>NSAppTransportSecurity</key>
 	<dict>
-		<key>NSAllowsArbitraryLoads</key>
-		<true/>
```
Comment (maxammann) [inline native/ios/Integreat/Info.plist:36]: I think we need this for HTTP traffic: https://developer.apple.com/documentation/bundleresources/information_property_list/nsapptransportsecurity/nsallowsarbitraryloads

**Review by maxammann — APPROVED** [review]
Looks good!

File: native/ios/Integreat/Info.plist (hunk 3 above)
Comment (klinzo) [reply native/ios/Integreat/Info.plist:36]: Hmm, react native upgrade helper says to remove it: https://react-native-community.github.io/upgrade-helper/?from=0.63.4&to=0.64.2 @maxammann

Comment (maxammann) [reply native/ios/Integreat/Info.plist:36]: Maybe RN had it previously by default. But I think we really need it. Unless there is an other option in XCode to allow HTTP traffic. By default it is forbidden on iOS and Android.

Comment (klinzo) [reply native/ios/Integreat/Info.plist:36]: HTTP is not working on Android either I think right now

File: native/android/app/build.gradle (hunk 1 above)
Comment (klinzo) [reply native/android/app/build.gradle:132]: @Taggotty could you try again please?

**Comment by klinzo** [comment]
Could someone also verify text in the location search on the landing page is visible?

**Review by f1sh1918 — APPROVED** [review]
Tested successfully on ios and android!
